spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestReadProjection.java

```diff
@@ -49,7 +50,7 @@ public abstract class TestReadProjection {
   protected abstract Record writeAndRead(
       String desc, Schema writeSchema, Schema readSchema, Record record) throws IOException;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir protected Path temp;
```

Comment:
because you're changing this file it seems you'd also have to update `TestSparkReadProjection`, otherwise CI won't pass.

Alternatively you could reintroduce
```
import org.junit.Assert;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;
```

but leave the AssertJ conversions in this file

Reply:
I chose to reintroduce the imports.
For that, I reversed the TempDir change, and changed 1 AssertJ back to Assert (added a TODO so I can change it in the future).
Otherwise, the imports were getting auto-removed from the IDE.